tests/common/tests.rs

```diff
@@ -916,3 +916,522 @@ async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
+	let root_token = socket.send_message_signin(USER, PASS, None, None, None).await.unwrap();
+	// Specify a namespace and database
+	socket.send_message_use(Some(NS), Some(DB)).await.unwrap();
+	// Setup the scope
+	socket
+		.send_message_query(
+			r#"
+			DEFINE SCOPE scope SESSION 1s
+				SIGNUP ( CREATE user SET email = $email, pass = crypto::argon2::generate($pass) )
+				SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+			;"#,
+		)
+		.await.unwrap();
+	// Create resource that requires a scope session to query
+	socket
+		.send_message_query(
+			r#"
+			DEFINE TABLE test SCHEMALESS
+				PERMISSIONS FOR select, create, update, delete WHERE $scope = "scope"
+			;"#,
+		)
+		.await.unwrap();
+	socket
+		.send_message_query(
+			r#"
+			CREATE test:1 SET working = "yes"
+			;"#,
+		)
+		.await.unwrap();
+	// Send SIGNUP command
+	let res = socket
+		.send_request(
+			"signup",
+			json!(
+				[{
+					"ns": NS,
+					"db": DB,
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}]
+			),
+		)
+		.await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	// Verify response contains no error
+	assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+	// Verify it returns a token
+	assert!(res["result"].is_string(), "result: {:?}", res);
+	let res = res["result"].as_str().unwrap();
+	assert!(res.starts_with("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9"), "result: {}", res);
+	// Authenticate using the token, which will expire soon
+	socket.send_request("authenticate", json!([res,])).await.unwrap();
+	// Check if the session is now authenticated
+	let res = socket.send_message_query("SELECT VALUE working FROM test:1").await.unwrap();
+	assert_eq!(res[0]["result"], json!(["yes"]), "result: {:?}", res);
+	// Wait two seconds for the session to expire
+	tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+	// Check if the session is now expired
+	let res = socket.send_request("query", json!(["SELECT VALUE working FROM test:1",])).await;
+	assert!(res.is_ok(), "result: {:?}", res);
+	let res = res.unwrap();
+	assert!(res.is_object(), "result: {:?}", res);
+	let res = res.as_object().unwrap();
+	assert_eq!(res["error"], json!({"code": -32000, "message": "There was a problem with the database: The session has expired"}));
+	// Test operations that SHOULD NOT work with an expired session
+	let operations_ko = vec![
+		socket.send_request("let", json!(["let_var", "let_value",])).await,
+		socket.send_request("set", json!(["set_var", "set_value",])).await,
+		socket.send_request("info", json!([])).await,
+		socket.send_request("select", json!(["tester",])).await,
+		socket
+			.send_request(
+				"insert",
+				json!([
+					"tester",
+					{
+						"name": "foo",
+						"value": "bar",
+					}
+				]),
+			)
+			.await,
```

Comment:
If we want to be webscale here, we can have these be futures and then `tokio::join!` them... but im fine with the way this is tbf

Reply:
As we discussed privately, some of the WS operations (e.g. `invalidate`) affect the state of the session in a way where they would need to be called in a specific order to prevent the test from becoming flaky. However, I agree that this logic can be implemented in a cleaner way and should be documented.

I have done that in [e3b5bb3](https://github.com/surrealdb/surrealdb/pull/3561/commits/e3b5bb33f3a2c4b50ea7cebe0a7fa46a66fa15a4).

Reply:
Since the futures are executed sequentially, I removed the use of `tokio::stream` in https://github.com/surrealdb/surrealdb/pull/3561/commits/025761e5f9e8132d8bdcf16aca88fca7a5940c7b.